src/Glpi/Form/AccessControl/ControlType/AllowListDropdown.php

```diff
@@ -63,6 +65,86 @@ public static function countUsersForCriteria(
         array $groups,
         array $profiles,
     ): array {
+        return [
+            'count' => self::countUsers($users, $groups, $profiles),
+            'link'  => self::computeSearchResultLink($users, $groups, $profiles),
+        ];
+    }
+
+    #[Override]
+    protected static function getAjaxUrl(): string
+    {
+        /** @var array $CFG_GLPI */
+        global $CFG_GLPI;
+
+        return $CFG_GLPI['root_doc'] . "/ajax/form/allowListDropdownValue.php";
+    }
+
+    #[Override]
+    protected static function getTypes(): array
+    {
+        return [
+            User::getType(),
+            Profile::getType(),
+            Group::getType(),
+        ];
+    }
+
+    protected static function countUsers(
```

Comment:
New method to count users more efficiently, the previous one used the search engine which take the current entities into account.
We want an accurate number here no matter what the current entity of the user.